ruma-events/src/room/message.rs

```diff
@@ -690,3 +759,235 @@ pub struct CustomEventContent {
+        }
+        MessageType::Emote(content) => {
+            format!("> * <{:?}> {}", original_message.sender, content.body)
+        }
+        MessageType::File(_) => {
+            format!("> <{:?}> sent a file.", original_message.sender)
+        }
+        MessageType::Image(_) => {
+            format!("> <{:?}> sent an image.", original_message.sender)
+        }
+        MessageType::Location(content) => {
+            format!("> <{:?}> {}", original_message.sender, content.body)
+        }
+        MessageType::Notice(content) => {
+            format!("> <{:?}> {}", original_message.sender, content.body)
+        }
+        MessageType::ServerNotice(content) => {
+            format!("> <{:?}> {}", original_message.sender, content.body)
+        }
+        MessageType::Text(content) => {
+            format!("> <{:?}> {}", original_message.sender, content.body)
+        }
+        MessageType::Video(_) => {
+            format!("> <{:?}> sent a video.", original_message.sender)
+        }
+        MessageType::_Custom(content) => {
+            format!(
+                "> <{:?}> {}",
+                original_message.sender,
+                content.data["body"].as_str().unwrap_or("")
+            )
+        }
+        #[cfg(feature = "unstable-pre-spec")]
+        MessageType::VerificationRequest(content) => {
+            format!("> <{:?}> {}", original_message.sender, content.body)
+        }
+    }
+}
+
+fn get_html_quote_fallback(original_message: &MessageEvent) -> String {
+    match &original_message.content.msgtype {
+        MessageType::Audio(_) => {
+            formatdoc! {"
+                <mx-reply>
+                    <blockquote>
+                        <a href=\"https://matrix.to/#/{room_id}/{event_id}\">In reply to</a>
+                        <a href=\"https://matrix.to/#/{sender}\">{sender}</a>
+                        <br />
+                        sent an audio file.
+                    </blockquote>
+                </mx-reply>",
+                room_id = original_message.room_id,
+                event_id = original_message.event_id,
+                sender = original_message.sender,
+            }
+        }
+        MessageType::Emote(content) => {
+            formatdoc! {"
+                <mx-reply>
+                    <blockquote>
+                        <a href=\"https://matrix.to/#/{room_id}/{event_id}\">In reply to</a>
+                        * <a href=\"https://matrix.to/#/{sender}\">{sender}</a>
+                        <br />
+                        {body}
+                    </blockquote>
+                </mx-reply>",
+                room_id = original_message.room_id,
+                event_id = original_message.event_id,
+                sender = original_message.sender,
+                body = if let Some(formatted) = &content.formatted {
+                    formatted.body.as_str()
+                } else {
+                    content.body.as_str()
+                }
```

Comment:
This same `if let` is used in a few places here, would be good to factor it out into a function, something like

```suggestion
                body = formatted_or_plain_body(&content.formatted, &content.body),
```

Reply:
Applied the suggestion.